lib/response.js

```diff
@@ -58,13 +58,18 @@ var charsetRegExp = /;\s*charset\s*=/;
 /**
  * Set status `code`.
  *
- * @param {Number} code
+ * @param {Number|String} code
```

Comment:
Hmm, Node.js documents their value as `number` and doesn't mention that strings should work (ref: https://nodejs.org/api/http.html#http_response_statuscode)

If we want to have documented support for strings, I think we should convert the status code to a number before passing it to Node.js, since the fact that strings works is just an undocumented implementation detail, and could potentially be changed in a future version 🤔 

Personally, I would prefer throwing on strings as well. That is, only accept `typeof code === 'number'`, since that seems to be the least surprising to me.

With the current code, the following code would work, but since it isn't documented we might accidentally break it in the future:

```
class CoolStatusCode {
  valueOf () {
    return 299
  }
}

// ...

res.status(new CoolStatusCode())
```

This would also be prevented with a `typeof code === 'number'` check.

-----

Anyways, this became a bit longer than I intended, hehe 😅. Bottom line is that I think that we shouldn't rely on undocumented features in Node.js core, and that I personally prefer being as strict as possible...

Reply:
I'm fine with making this change. I figured it was possible folks are already passing strings, whether they know it or not, since it just works currently. On the original PR folks were pro throwing on strings, I can revert it back to that behavior.

Reply:
To be clear, I have addressed this suggestion and the PR adds throwing on any string value passed.